src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts

```diff
@@ -375,6 +376,25 @@ export class InlineChatController implements IEditorContribution {
 
 		this._sessionStore.clear();
 
+		this._sessionStore.add(this._zone.value.widget.onRequestWithoutIntentDetection(async () => {
+			options.withIntentDetection = false;
+
+			// undo changes
+			if (this._activeSession && this._activeSession.lastExchange && this._strategy) {
+				const { lastExchange } = this._activeSession;
+				if (lastExchange.response instanceof ReplyResponse) {
+					try {
+						this._ignoreModelContentChanged = true;
+						await this._strategy.undoChanges(lastExchange.response.modelAltVersionId);
+					} finally {
+						this._ignoreModelContentChanged = false;
+					}
+				}
+			}
+
+			this.acceptInput();
```

Comment:
Maybe it is better to dispatch the RERUN_INPUT message after setting the option (L380). With that you don't need to undo the changes because it is done for you (see L512 following)